packages/product-components/src/components/SelectAssetModal/AssetItem.tsx

```diff
@@ -11,43 +11,67 @@ const ClickableContainer = styled.div`
         background: ${({ theme }) => theme.backgroundTertiaryPressedOnElevation0};
     }
 `;
+
 const IconWrapper = styled.div`
     cursor: pointer;
 `;
 
+const TextWrapper = styled.div`
+    overflow: hidden;
+    text-overflow: ellipsis;
+`;
+
+const BadgeWrapper = styled.div`
+    flex: none;
+`;
+
 type AssetItemProps = {
-    handleClick: (selectedAsset: string) => void;
+    cryptoId: string;
     name: string;
     symbol: string;
-    badge: string;
+    badge: string | undefined;
+    logo: JSX.Element;
     isFavorite?: boolean;
+    handleClick: (selectedAsset: string) => void;
     onFavoriteClick?: (isFavorite: boolean) => void;
 };
+
 export const AssetItem = ({
-    handleClick,
+    cryptoId,
     name,
     symbol,
     badge,
+    logo,
     isFavorite = false,
+    handleClick,
     onFavoriteClick,
 }: AssetItemProps) => {
     const theme = useTheme();
 
-    const handleFavoriteClick = () => onFavoriteClick?.(isFavorite);
+    const handleFavoriteClick = onFavoriteClick ? () => onFavoriteClick(isFavorite) : undefined;
 
     return (
-        <ClickableContainer onClick={() => handleClick(symbol)}>
-            <Row gap={spacings.sm} margin={{ right: spacings.md }}>
-                <AssetLogo size={24} coingeckoId="bitcoin" placeholder="btc" />
+        <ClickableContainer onClick={() => handleClick(cryptoId)}>
+            <Row gap={spacings.sm}>
+                {logo}
                 <Column flex="1" alignItems="stretch">
-                    <Text typographyStyle="body">{name}</Text>
+                    <Row gap={spacings.xs} alignItems="center">
+                        <TextWrapper>
+                            <Text typographyStyle="body" textWrap="nowrap">
+                                {name}
+                            </Text>
+                        </TextWrapper>
+                        {badge && (
+                            <BadgeWrapper>
+                                <Badge>{badge}</Badge>
+                            </BadgeWrapper>
+                        )}
+                    </Row>
                     <Text typographyStyle="hint" variant="tertiary">
                         {symbol}
```

Comment:
In send form, we plan to add this token select / modal also with contract addresses. Have you considered that too? Does it make sense?
cc @Hermez-cz

Reply:
![Screenshot 2024-09-27 at 16 24 12](https://github.com/user-attachments/assets/44b213fe-304a-4243-a9c5-8433a9f19c12)

Reply:
![Screenshot 2024-09-27 at 16 34 09](https://github.com/user-attachments/assets/b2b7c814-0fb5-4b30-8e6e-fee0b96549ec)

Reply:
Might make sense for the same reason as in Send, that is being able to distinguish unique tokens in a verifiable way. It might be too fuzzy in such a long list. But we might consider showing the contract address once a token has been selected. What do you think @marek-andrsk ?